test/Microsoft.VisualStudio.Editor.Razor.Test/DefaultTagHelperCompletionServiceTest.cs

```diff
@@ -838,6 +838,75 @@ public void GetElementCompletions_EnsuresDescriptorsHaveSatisfiedParent()
             AssertCompletionsAreEquivalent(expectedCompletions, completions);
         }
 
+        [Fact]
+        public void GetElementCompletions_NoContainingParentTag_DoesNotGetCompletionForRuleWithParentTag()
+        {
+            // Arrange
+            var documentDescriptors = new[]
+            {
+                TagHelperDescriptorBuilder.Create("Tag1", "TestAssembly")
+                    .TagMatchingRuleDescriptor(rule => rule.RequireTagName("tag-b"))
+                    .TagMatchingRuleDescriptor(rule => rule.RequireTagName("tag-a.1").RequireParentTag("tag-a"))
```

Comment:
Suggest using more descriptive names like `parent`, `child` here. It'll be easier to visualize.